_posts/2019/2019-07-25-how-to-generate-mac-addresses.md

```diff
@@ -0,0 +1,113 @@
+A classic (IPv4 referring) MAC address has 48 bits or 6 bytes, and is expressed in hexadecimal digits separated by colons between each of the 6 octets.
+
+If you divide these 6 octets by two, the first 3 are the so-called Organizationally Unique Identifiers (OUI), and the last 3 refer to a manufacturer-specific number.
+Important here are the first two bits of the first byte.
+Let's call them b0 and b1. 
+
+The b0 bit states if this MAC address is a unicast (0) or a [multicast](https://en.wikipedia.org/wiki/Multicast_address#Ethernet){:target="_blank"} (1) address.
+The b1 bit states if a MAC address is globally unique (0) or locally administered (1).
+
+## Some math
+
+As we now know that the first 3 bytes are reserved, let's call them the MAC prefix here.
+We assume that these 3 bytes are fixed for now.
+Using only the last 3 bytes we can generate up to 16^6 or 2^24 addresses which is exactly the size of a "/8 sized network.
+You may think now "Hey that's exactly the size of the biggest usable local network".
+
+There are three private IPv4 address ranges specified in [RFC1918](https://tools.ietf.org/html/rfc1918){:target="_blank"}:
+
+- 10.0.0.0 - 10.255.255.255  (10/8 prefix)
+- 172.16.0.0 - 172.31.255.255  (172.16/12 prefix)
+- 192.168.0.0 - 192.168.255.255 (192.168/16 prefix)
+
+## Local and global MAC addresses
+
+Generating only one MAC prefix allows us to generate enough addresses for the biggest assignable private network. 
+Cool thing, and we know that if we want to use it locally we only have to set the second bit in the first byte of this mac address to 1.
+
+You also have the option to [register](https://standards.ieee.org/products-services/regauth/index.html){:target="_blank"} a MAC prefix (with b1=0) for your own purposes.
+The registration is made by IEEE and costs around 1000$ - 3000$.
+IEEE also publishes a list of already existing [OUIs](http://standards-oui.ieee.org/oui.txt){:target="_blank"}.
+
+## Mac address prefix generation
+
+To simply generate a MAC address prefix, we can have the following options for the first byte:
+
+```
+bbbbbb10 , bbbbbb01, bbbbbb00, bbbbbb11
+```
+where b ∊ {0,1} can be set at random.
+
+The first one (10) is a locally administered unicast address, and it's the most common use case if you need a local MAC address.
+The second one (01) is a globally unique multicast address, and very rare, since multicast addresses are not used very often.
+Our third one (00) is a very typical representative of its species: the global unique unicast address.
+This one is the most common OUI used by almost any hardware producing company (or even software-related when it comes to virtualization). 
+Almost everyone who needs a OUI uses this pattern.
+Last but not least (11) is the locally administered multicast, and also a rare one.
+
+MAC addresses are commonly written in hex and not binary representation, so let's take a look how our candidates look in hex.
```

Comment:
The first part of this sentence reads, as if something is missing there.

Reply:
I think it's fine if you just add an "in" to clarify that a bit more.

--> MAC addresses are commonly written in hex and not IN binary representation,

Reply:
done